### PR TITLE
Rewrite libxml2 and libxslt phases

### PR DESCRIPTION
**Pull request by hmelder**

### CODE REVIEW

**Comment by triplef** [comment]
Looks Base is trying to link `libxml-2.0.lib`, even though the library installed is named `xml2.lib`. What does it say in the pkgconfig file?

**Comment by hmelder** [comment]
> Looks Base is trying to link `libxml-2.0.lib`, even though the library installed is named `xml2.lib`. What does it say in the pkgconfig file?

GNUstep x64 Debug:
image

It is probably another variable to be patched in the libxml2 CMakeLists.txt. I'm reluctant to patching, but the problem is that in order for libxslt to find the libxml2 library, libxml2 generates cmake modules during installation. This conflicts with the behaviour of manually copying the static library, headers, and generation of the pc file.

I got everything building with the existing phases + a small pkgconfig patch in the libxslt phase.

**Comment by hmelder** [comment]
But the configuration is correct:
https://github.com/GNOME/libxml2/blob/c21e9cd5d955e4d8afa514e1f7736ce6a9bb8f2e/CMakeLists.txt#L657

File: phases/17-libxslt.bat [hunk 1]
```diff
@@ -51,4 +51,4 @@ xcopy /Y /F /S "%SRCROOT%\%PROJECT%\libxslt\*.h" "%INSTALL_PREFIX%\include\libxs
 xcopy /Y /F /S "%SRCROOT%\%PROJECT%\libexslt\*.h" "%INSTALL_PREFIX%\include\libexslt\" || exit /b 1
 
 :: write pkgconfig file
-call "%~dp0\..\scripts\common.bat" write_pkgconfig libxslt %TAG% -DLIBXSLT_STATIC -lxslt libxml-2.0 || exit /b 1
+call "%~dp0\..\scripts\common.bat" write_pkgconfig libxslt %TAG% -DLIBXSLT_STATIC -lxslt -lxml2 || exit /b 1
```
Comment (triplef) [inline phases/17-libxslt.bat:54]: I think the last 3 parameters must be in quotes for the write_pkgconfig script to work correctly.

**Pull request closed without merging**